libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m

```diff
@@ -333,9 +338,71 @@ - (SFRestRequest *)requestForSearchResultLayout:(NSString*)objectList {
+    NSMutableArray *requestsArrayJson = [NSMutableArray new];
+    for (SFRestRequest *request in requests) {
+        NSMutableDictionary<NSString *, id> *requestJson = [NSMutableDictionary new];
+        requestJson[@"method"] = [SFRestRequest httpMethodFromSFRestMethod:request.method];
+        // queryParams belong in url
+        if (request.method == SFRestMethodGET || request.method == SFRestMethodDELETE) {
+            requestJson[@"url"] = [NSString stringWithFormat:@"%@%@", request.path, [self toQueryString:request.queryParams]];
+        }
+        // queryParams belongs in body
+        else {
+            requestJson[@"url"] = request.path;
+            requestJson[@"richInput"] = request.queryParams;
+        }
+        [requestsArrayJson addObject:requestJson];
+    }
+    NSMutableDictionary<NSString *, id> *batchRequestJson = [NSMutableDictionary new];
+    batchRequestJson[@"batchRequests"] = requestsArrayJson;
+    batchRequestJson[@"haltOnError"] = [NSNumber numberWithBool:haltOnError];
+
+    NSString *path = [NSString stringWithFormat:@"/%@/composite/batch", self.apiVersion];
+
+    return [SFRestRequest requestWithMethod:SFRestMethodPOST path:path queryParams:batchRequestJson];
+}
+
+- (SFRestRequest *) compositeRequest:(NSArray<SFRestRequest*>*) requests refIds:(NSArray<NSString*>*)refIds allOrNone:(BOOL) allOrNone {
+    NSMutableArray *requestsArrayJson = [NSMutableArray new];
+    for (int i=0; i<requests.count; i++) {
+        SFRestRequest *request = requests[i];
+        NSString *refId = refIds[i];
+        NSMutableDictionary<NSString *, id> *requestJson = [NSMutableDictionary new];
+        requestJson[@"referenceId"] = refId;
+        requestJson[@"method"] = [SFRestRequest httpMethodFromSFRestMethod:request.method];
+        // queryParams belong in url
+        if (request.method == SFRestMethodGET || request.method == SFRestMethodDELETE) {
+            requestJson[@"url"] = [NSString stringWithFormat:@"%@%@%@", request.endpoint, request.path, [self toQueryString:request.queryParams]];
+        }
+        // queryParams belongs in body
+        else {
+            requestJson[@"url"] = [NSString stringWithFormat:@"%@%@", request.endpoint, request.path];
+            requestJson[@"body"] = request.queryParams;
+        }
+        [requestsArrayJson addObject:requestJson];
+    }
+    NSMutableDictionary<NSString *, id> *compositeRequestJson = [NSMutableDictionary new];
+    compositeRequestJson[@"compositeRequest"] = requestsArrayJson;
+    compositeRequestJson[@"allOrNone"] = [NSNumber numberWithBool:allOrNone];
 
-    return [httpDateFormatter stringFromDate:date];
+    NSString *path = [NSString stringWithFormat:@"/%@/composite", self.apiVersion];
+
+    return [SFRestRequest requestWithMethod:SFRestMethodPOST path:path queryParams:compositeRequestJson];
 }
+
+- (NSString *)toQueryString:(NSDictionary *)components {
```

Comment:
I'm encoding the value by calling stringByURLEncoding.

Ideally the request factory methods should construct the right url or request body (like we do on Android), so that I wouldn't have to do anything in the batch or composite method.